server/controllers/orderItemController.js

```diff
@@ -71,6 +71,53 @@ module.exports = {
         }
     },
 
-    //future additions include the ability to update an order, delete an order
+    updateOrder: async (req, res) => {
+        if(!mongoose.Types.ObjectId.isValid(req.params.id)){
+            return res.status(400).json({ error: 'Enter a valid Order id' });
+        }
+        let order = await OrderItemModel.findById({ _id : req.params.id});
+        if(!order) return res.status(400).json({ error: 'The Order is inexistent' });
+        let pizza = await PizzaModel.findOne({ _id: req.body.pizza_id });
+        if(!pizza) return res.status(400).json({ error: 'The Pizza is inexistent' })
+        let tempTotal = pizza.price * req.body.quantity;
+        let discount = 0;
+        if(tempTotal > 50 && tempTotal < 100){
+            discount = (5/100) * tempTotal;
+        }else if(tempTotal >= 100){
+            discount = (10/100) * tempTotal;
+        }
+        let totalAmount = tempTotal - discount;
+        await OrderItemModel.findByIdAndUpdate(req.params.id, {
+            pizza_id : req.body.pizza_id,
+            quantity: req.body.quantity,
+            totalAmount
+        }, {new: true},  (err, order) => {
+            if(err) return res.status(500).json({ error: err.message })
+            return res.status(200).json({ 
+                order : {
+                    pizza_id: order.pizza_id,
+                    quantity: order.quantity,
+                    totalAmount: tempTotal - discount,
+                    discount
+
+                }
+            });
+        })
+    },
 
+    deleteOrder: async (req, res) => {

```

Comment:
Looking at thiis method, I can see that I can delete anyones order
Or is it something happening higher up in the levels to secure that I cannot delete anothers order?

Reply:
Authentication is happening higher up.

Reply:
@mutebironald but is it protected towards the specific orders? Its fine that it is protected, but does it protect a set of orders?

Reply:
So currently, it isn't protected towards specific orders but I will extend that functionality later on.

Reply:
@mutebironald so not need to, you can explain how you would have done it and that is fine. We are overtime now :D

Reply:
@vongohren I would have a `created_by` attribute on the `OrderItems` model. This attribute will store the user who made the order. On a delete request, I check if the current user is the one who created this order, if they created the order, I proceed with deletion.

Reply:
Sounds good